File: services/dotnet/BabelServices/Babel.Services.Library/ProfileService.cs
using System;
using System.Collections.Generic;
using System.ServiceModel;

using Babel.Services.Repositories;
using Babel.Services.Domain;
using Babel.Services.Filters;


namespace Babel.Services
{
    [ServiceBehavior(Namespace = "http://babylon.com/services")]
    public class ProfileService : IProfileService
    {
        private IProfileRepository _repository;

        public Guid AddProfile(Profile profile)
        {
            Repository.Add(profile);

            return profile.ID;
        }

        public Guid CreateProfile(string username, string password, string email, string name, string surname)
        {
            Profile newProfile = new Profile();

            newProfile.Username = username;
            newProfile.Password = password;
            newProfile.Email = email;
            newProfile.Name = name;
            newProfile.Surname = surname;

            Repository.Add(newProfile);

            return newProfile.ID;
        }

        public Profile GetProfile(string username, string password)
        {
            return Repository.GetProfileByCredentials(username, password);
        }

        public Profile GetProfile(Guid id)
        {
            return Repository.GetProfileByID(id);
        }

        public IList<Profile> GetAllProfiles()
        {
            return Repository.GetAllProfiles();
        }

        public void ModifyProfile(Profile profile)
        {
            Repository.Update(profile);
        }

        public void DeleteProfile(Guid id)
        {
            Profile profile = Repository.GetProfileByID(id);
            Repository.Remove(profile);
        }

        public IList<Profile> SearchProfiles(ProfileFilter filter)
        {
            return Repository.SearchProfiles(filter);
        }

        public void AddContact(Guid id, Guid contactID)
        {
            Repository.AddContact(id, contactID);
        }

        public void AddContacts(Guid id, IList<Guid> contactIDs)
        {
            Repository.AddContacts(id, contactIDs);
        }

        public IList<Profile> GetContacts(Guid id)
        {
            return Repository.GetContacts(id);
        }

        public void RemoveContact(Guid id, Guid contactID)
        {
            Repository.RemoveContact(id, contactID);
        }

        public void RemoveAllContacts(Guid id)
        {
            Repository.RemoveAllContacts(id);
        }

        private IProfileRepository Repository
        {
            get
            {
                if (_repository == null)
                {
                    _repository = new Babel.Services.Repositories.ProfileRepository();
                }
                return _repository;
            }
        }
    }
}

File: services/java/Babylon.Services/Babylon.Services.Metro/src/java/Babylon/Services/Metro/ProfileService.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package Babylon.Services.Metro;

import javax.jws.WebMethod;
import javax.jws.WebParam;
import javax.jws.WebService;

import Babylon.Services.Adapters.ProfileServiceAdapter;
import Babylon.Services.Proxies.ProfileService.ArrayOfProfile;
import Babylon.Services.Proxies.ProfileService.ArrayOfguid;
import Babylon.Services.Proxies.ProfileService.Profile;

/**
 *
 * @author guille
 */
@WebService(serviceName = "ProfileService")
public class ProfileService {

    /**
     * AddContact web service operation
     */
    @WebMethod(operationName = "AddContact")
    public void addContact(
            @WebParam(name = "id") String id,
            @WebParam(name = "name") String contactId)
    {
        
        ProfileServiceAdapter adapter = new ProfileServiceAdapter();
        adapter.addContact(id, contactId);
    }
    
    @WebMethod(operationName = "AddContacts")
    public void addContacts(
            @WebParam(name = "id") String id,
            @WebParam(name = "contactIDs") ArrayOfguid contactIDs)
    {
        
        ProfileServiceAdapter adapter = new ProfileServiceAdapter();
        adapter.addContacts(id, contactIDs);
    }
    
    @WebMethod(operationName = "AddProfile")
    public String addProfile(
            @WebParam(name = "profile") Profile profile)
    {
        ProfileServiceAdapter adapter = new ProfileServiceAdapter();
        return adapter.addProfile(profile);
    }
    
    @WebMethod(operationName = "CreateProfile")
    public String createProfile(
            @WebParam(name = "username") String username,
            @WebParam(name = "password") String password,
            @WebParam(name = "email") String email,
            @WebParam(name = "name") String name,
            @WebParam(name = "surname") String surname)
    {
        ProfileServiceAdapter adapter = new ProfileServiceAdapter();
        return adapter.createProfile(username, password, email, name, surname);
    }
    
    @WebMethod(operationName = "DeleteProfile")
    public void deleteProfile(
            @WebParam(name = "id") String id)
    {
        ProfileServiceAdapter adapter = new ProfileServiceAdapter();
        adapter.deleteProfile(id);
    }
    
    @WebMethod(operationName = "GetAllProfiles")
    public ArrayOfProfile getAllProfiles() {
        ProfileServiceAdapter adapter = new ProfileServiceAdapter();
        return adapter.getAllProfiles();
    }
    
    @WebMethod(operationName = "GetContacts")
    public ArrayOfProfile getContacts(
            @WebParam(name = "id") String id) {
        ProfileServiceAdapter adapter = new ProfileServiceAdapter();
        return adapter.getContacts(id);
    }
    
    @WebMethod(operationName = "GetProfileByCredentials")
    public Profile getProfileByCredentials(
            @WebParam(name = "username") String username,
            @WebParam(name = "password") String password)
    {
        ProfileServiceAdapter adapter = new ProfileServiceAdapter();
        return adapter.getProfileByCredentials(username, password);
    }
    
    @WebMethod(operationName = "GetProfileByID")
    public Profile getProfileByID(
            @WebParam(name = "id") String id)
    {
        ProfileServiceAdapter adapter = new ProfileServiceAdapter();
        return adapter.getProfileByID(id);
    }
    
    @WebMethod(operationName = "ModifyProfile")
    public void modifyProfile(
            @WebParam(name = "profile") Profile profile)
    {
        ProfileServiceAdapter adapter = new ProfileServiceAdapter();
        adapter.modifyProfile(profile);
    }
    
    @WebMethod(operationName = "RemoveAllContacts")
    public void removeAllContacts(
            @WebParam(name = "id") String id)
    {
        ProfileServiceAdapter adapter = new ProfileServiceAdapter();
        adapter.removeAllContacts(id);
    }
    
    @WebMethod(operationName = "RemoveContact")
    public void removeContact(
            @WebParam(name = "id") String id,
            @WebParam(name = "contactID") String contactID)
    {
        ProfileServiceAdapter adapter = new ProfileServiceAdapter();
        adapter.removeContact(id, contactID);
    }
    
}
